tests/external_executor/external_executor.py

```diff
@@ -421,6 +427,105 @@ def subscribe(event_name):
     return network
 
 
+@reqs.description("Test index API")
+def test_index_api(network, args):
+    primary, _ = network.find_primary()
+
+    def add_kv_entries(network):
+        logging_executor = LoggingExecutor(primary)
+        supported_endpoints = logging_executor.supported_endpoints
+        credentials = register_new_executor(
+            primary, network, supported_endpoints=supported_endpoints
+        )
+        logging_executor.credentials = credentials
+        log_id = 14
+        with executor_thread(logging_executor):
+            with primary.client() as c:
+                for _ in range(3):
+                    r = c.post(
+                        "/app/log/public",
+                        {"id": log_id, "msg": "hello_world_" + str(log_id)},
+                    )
+                    assert r.status_code == 200
+                    log_id = log_id + 1
+
+    add_kv_entries(network)
+
+    credentials = register_new_executor(primary, network)
+
+    with grpc.secure_channel(
+        target=f"{primary.get_public_rpc_host()}:{primary.get_public_rpc_port()}",
+        credentials=credentials,
+    ) as channel:
+        data = queue.Queue()
+        subscription_started = threading.Event()
+
+        def InstallandSub():
+            sub_credentials = register_new_executor(primary, network)
+
+            with grpc.secure_channel(
+                target=f"{primary.get_public_rpc_host()}:{primary.get_public_rpc_port()}",
+                credentials=sub_credentials,
+            ) as subscriber_channel:
+                in_stub = IndexService.IndexStub(subscriber_channel)
+                for work in in_stub.InstallAndSubscribe(
+                    Index.IndexInstall(
+                        strategy_name="TestStrategy",
+                        map_name="public:records",
+                        data_structure="MAP",
```

Comment:
Shouldn't this be an enum value rather than a string?

Reply:
My bad, changed it